Microsoft.WindowsAzure.Storage/tests/cloud_table_test.cpp

```diff
@@ -2689,11 +2689,24 @@ SUITE(Table)
             web::json::value input_document = web::json::value::object(fields);
             utility::string_t message = input_document.serialize();
             web::json::value output_document = web::json::value::parse(message);
-            
+
             CHECK(output_document.is_object());
             CHECK(output_document.as_object().find(_XPLATSTR("DoubleProperty")) != output_document.as_object().cend());
             CHECK_EQUAL(web::json::value::value_type::Number, output_document.as_object().find(_XPLATSTR("DoubleProperty"))->second.type());
-            CHECK(output_document.as_object().find(_XPLATSTR("DoubleProperty"))->second.is_double());
+
+            double intpart;
+            double fracpart = modf(double_value, &intpart);
+            if (fracpart == 0.0)
+            {
+                // if we happen to generate a whole number, it will be reported as an integer
+                CHECK(output_document.as_object().find(_XPLATSTR("DoubleProperty"))->second.is_integer());
+            }
+            else
+            {
+                CHECK(output_document.as_object().find(_XPLATSTR("DoubleProperty"))->second.is_double());
+            }
+
+            // even if we generate a whole number, it should have the same double value
```

Comment:
Not sure I understand. Do you want me change the assertion to just be that is `is_number()`? Or do you want me to remove the commit and leave the test broken?